### PR TITLE
:sparkles: 266: Added option to disable the space between the header and the body

### PR DESCRIPTION
**Pull request by Scoty**
when using heroStyle: background.

This is one possible fix/extension for #266

### CODE REVIEW

**Comment by nunocoracao** [comment]
Hello @Scoty can I suggest a couple of changes, I can make them later on top of your PR if you don't have the time let me know. 

- Change parameter name to be `layoutBackgroundHeaderSpace`
- Add a global version of the parameter
- Add a front-matter version of the parameter
- Make sure the code in the partial goes through all the layers of the flag - global > list/article > front-matter
- Add the 4 different parameters to the docs (global/list/article in configurations, front-matter in front-matter doc)

Let me know if you can help here otherwise I'll try to do it once I have time

**Comment by Scoty** [comment]
I will do the changes, but I might not have the time today :)

**Comment by nunocoracao** [comment]
Really appreciate the help @Scoty

File: layouts/partials/hero/background.html [hunk 1]
```diff
@@ -4,7 +4,9 @@
 {{- if not $featured }}{{ with .Site.Params.defaultBackgroundImage }}{{ $featured = resources.Get . }}{{ end }}{{ end -}}
 {{- with $featured -}}
 {{ with .Resize "1200x" }}
+{{ if $.Site.Params.article.layoutBackgroundMoreSpaceBetweenHeaderAndBody | default true }}
```
[the remark this answers is not in the document]
Comment (Scoty) [reply layouts/partials/hero/background.html:7]: One question here - what is the way to know that the background.html partial was added to a list or an article? Cause that is the info, I need to do the check for the proper parameter value (either `Params .list.xxx` or `Params .article.xxx`)

[the remark this answers is not in the document]
Comment (Scoty) [reply layouts/partials/hero/background.html:7]:
```go
{{ if or (and ($.Site.Params.article.layoutBackgroundHeaderSpace | default true) (.isPage) )
         (and ($.Site.Params.list.layoutBackgroundHeaderSpace | default true) (.isNode) ) }}
```

Tried the above, but it throws an error:

```
execute of template failed: template: partials/hero/background.html:7:82: executing "partials/hero/background.html" at <.isPage>: can't evaluate field isPage in type images.ImageResource
```

Will dig deeper tomorrow...

**Comment by Scoty** [comment]
Btw if #245 is going to be implemented soon - this change might not be required as the concept of a page's `hero` and `background` with be separated. Therefore when adding a `background` to a page and not having `hero` the empty `hero` div will not appear.

**Comment by nunocoracao** [comment]
Not really it still might make sense to leave both options... and also, we could do it in stages. Merge this right now for you and others while we investigate the other change more.

**Comment by Scoty** [comment]
Added the list, article, and shortcode config. Updated the relevant docs.

**Comment by Scoty** [comment]
@nunocoracao anything more left to implement? The one thing I did not implement was the global flag, but as both `heroStyle` and `layoutBackgroundBlur` did not have one, I prefer to leave the most generic ones to be article and list.

**Comment by nunocoracao** [comment]
Sorry @Scoty busy week. Merging this to `dev` to test with all the other changes before making a new release

**Pull request merged**